src/util/api-request.ts

```diff
@@ -0,0 +1,55 @@
+import { Cleaner } from 'cleaners'
+import { EdgeFetchFunction, EdgeLog } from 'edge-core-js'
```

Comment:
We should provide our own log type locally to this repo, instead of using `EdgeLog`, since `EdgeLog` has added new stuff for bug reporting. The new type would still be a subset of `EdgeLog`, though, so users of the core can still pass `EdgeLog` instances in and they will be acceptable.

Reply:
How about `Pick<EdgeLog, 'error' | 'warn'>`?